.github/workflows/build.yml

```diff
@@ -35,10 +41,9 @@ jobs:
       with:
         java-version: '11'
         distribution: 'temurin'
-        cache: gradle
 
-    - name: Grant execute permission for gradlew
-      run: chmod +x gradlew
-      
-    - name: Build and test with Gradle
-      run: ./gradlew build
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2
+
+    - name: Gradle Check
+      run: ./gradlew check
```

Comment:
I changed 'build' to 'check', because the build output wasn't used so it should be faster - but maybe you want to test the build too, so I can revert it you want?

Reply:
Hi @aSemy shall we run this `check` as a part of the lint job earlier?

Reply:
nvm, check already runs `apiCheck` - I'll remove the earlier job in a separate commit

Reply:
> but maybe you want to test the build too, so I can revert it you want?

Yeah, I want to test and store the build artifacts in staging and probably publish the snapshots on sonatype. But those will be a different job